dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsAPIImplTest.java

```diff
@@ -418,6 +432,135 @@ public void Test_Secret_Destroy_Method() {
+                new AppTestCase("", "","", "", emptyParams),
+                new AppTestCase("any-key", "","", "",  emptyParams),
+                new AppTestCase("any-key", "any-name","", "", emptyParams),
+                new AppTestCase("any-key", "any-name","desc", "", emptyParams),
+                new AppTestCase("any-key", "any-name","desc", "icon", emptyParams),
+                //Name too large.
+                new AppTestCase(RandomStringUtils.randomAlphanumeric(AppsAPIImpl.DESCRIPTOR_KEY_MAX_LENGTH + 1), "any-name","desc", "icon", emptyParams),
+                //Key-too large
+                new AppTestCase("any-key", RandomStringUtils.randomAlphanumeric(AppsAPIImpl.DESCRIPTOR_NAME_MAX_LENGTH + 1),"desc", "icon", emptyParams),
+                //The following test paramDefinition
+                //Null type  is not allowed.
+                new AppTestCase("any-key", "any-name", "desc", "icon", ImmutableSortedMap.of(
+                  "p1", ParamDescriptor.newParam("", true, null, "", "", true)
+                )),
+                //Hidden bool param is not allowed.
+                new AppTestCase("any-key", "any-name", "desc", "icon", ImmutableSortedMap.of(
+                  "p1", ParamDescriptor.newParam("", true, Type.BOOL, "label", "hint", true)
+                )),
+                //emptyLabel
+                new AppTestCase("any-key", "any-name", "desc", "icon", ImmutableSortedMap.of(
+                  "p1", ParamDescriptor.newParam("v1", true, Type.STRING, "", "", true)
+                )),
+                //emptyHint
+                new AppTestCase("any-key", "any-name", "desc", "icon", ImmutableSortedMap.of(
+                 "p1", ParamDescriptor.newParam("v1", true, Type.STRING, "label", "", true)
+                )),
+                //Required param with null default
+                new AppTestCase("any-key", "any-name", "desc", "icon", ImmutableSortedMap.of(
+                        "p1", ParamDescriptor.newParam("null", false, Type.STRING, "label", "hint", true)
+                )),
+                //non parsable to bool string
+                new AppTestCase("any-key", "any-name", "desc", "icon", ImmutableSortedMap.of(
+                        "p1", ParamDescriptor.newParam("lol", false, Type.BOOL, "label", "hint", true)
+                ))
+        };
+    }
+
+    /**
+     * Test AppsAPI#createAppDescriptor input validation.
+     * Given scenario: Passing Params that normally wouldn't break the validation rules.
+     * Expected Result: No DotDataValidationException is raised.
+     */
+    @Test
+    @UseDataProvider("getValidExceptionFreeTestCases")
+    public void Test_App_Descriptor_Validation_Exception_Free(final AppTestCase testCase)
+            throws IOException, DotDataException, DotSecurityException {
+        assertNotNull(evaluateAppTestCase(testCase));
+    }
+
+
+    @DataProvider
+    public static Object[] getValidExceptionFreeTestCases() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5358201)